apps/server/src/modules/task/service/submission.service.ts

```diff
@@ -26,4 +31,104 @@ export class SubmissionService {
 
 		await this.submissionRepo.delete(submission);
 	}
+
+	async deleteSubmissionsByUserId(userId: EntityId): Promise<DomainOperation> {
+		this.logger.info(
+			new DataDeletionDomainOperationLoggable(
+				'Deleting data from Submissions',
+				DomainName.SUBMISSIONS,
+				userId,
+				StatusModel.PENDING
+			)
+		);
+		const submissions = await this.submissionRepo.findAllByUserId(userId);
+		const submissionsToDelete: Submission[] = [];
+
+		if (submissions[1] > 0) {
+			submissions[0].forEach((submission) => {
+				if (submission.courseGroup === null && submission.teamMembers.length === 1) {
```

Comment:
Please move `submission.courseGroup === null && submission.teamMembers.length === 1` into some `Submission` class method to make it easier to read (e.g. `isOwnedByUser()` as I think submission can be considered as owned by the user if it's not a group submission and there's only a single user in this `teamMembers` list). Please also use `filter()` method then instead of this loop.